packages/api/src/clickhouse/index.ts

```diff
@@ -904,6 +912,588 @@ export const getMetricsChart = async ({
+        _string_attributes
+      FROM ??
+      WHERE name = ?
+      AND data_type = ?
+      AND (?)
+      GROUP BY name, _string_attributes, timestamp
+      ORDER BY timestamp ASC
+    `.trim(),
+    [granularity, tableName, name, dataType, SqlString.raw(whereClause)],
+  );
+
+  const query = SqlString.format(
+    `
+      WITH metrics AS (?)
+      SELECT ?
+      FROM metrics
+      GROUP BY group, ts_bucket
+      ORDER BY ts_bucket ASC
+      ${
+        granularity != null
+          ? `WITH FILL
+        FROM toUnixTimestamp(toStartOfInterval(toDateTime(?), INTERVAL ?))
+        TO toUnixTimestamp(toStartOfInterval(toDateTime(?), INTERVAL ?))
+        STEP ?`
+          : ''
+      }
+    `,
+    [
+      SqlString.raw(
+        isRate
+          ? rateMetricSource
+          : // Max/Min aggs are the same for both Sum and Gauge metrics
+          dataType === 'Sum' && aggFn != AggFn.Max && aggFn != AggFn.Min
+          ? sumMetricSource
+          : gaugeMetricSource,
+      ),
+      SqlString.raw(selectClause.join(',')),
+      ...(granularity != null
+        ? [
+            startTime / 1000,
+            granularity,
+            endTime / 1000,
+            granularity,
+            ms(granularity) / 1000,
+          ]
+        : []),
+    ],
+  );
+
+  return {
+    query,
+    hasGroupBy,
+  };
+};
+
+const buildEventSeriesQuery = async ({
+  aggFn,
+  endTime,
+  field,
+  granularity,
+  groupBy,
+  maxNumGroups,
+  propertyTypeMappingsModel,
+  q,
+  sortOrder,
+  startTime,
+  tableVersion,
+  teamId,
+}: {
+  aggFn: AggFn;
+  endTime: number; // unix in ms,
+  field?: string;
+  granularity: string | undefined; // can be undefined in the number chart
+  groupBy: string;
+  maxNumGroups: number;
+  propertyTypeMappingsModel: LogsPropertyTypeMappingsModel;
+  q: string;
+  sortOrder?: 'asc' | 'desc';
+  startTime: number; // unix in ms
+  tableVersion: number | undefined;
+  teamId: string;
+}) => {
+  if (isRateAggFn(aggFn)) {
+    throw new Error('Rate is not supported in logs chart');
+  }
+
+  const tableName = getLogStreamTableName(tableVersion, teamId);
+  const whereClause = await buildSearchQueryWhereCondition({
+    endTime,
+    propertyTypeMappingsModel,
+    query: q,
+    startTime,
+  });
+
+  if (field == null && aggFn !== AggFn.Count) {
+    throw new Error(
+      'Field is required for all aggregation functions except Count',
+    );
+  }
+
+  const selectField =
+    field != null
+      ? buildSearchColumnName(propertyTypeMappingsModel.get(field), field)
+      : '';
+
+  const hasGroupBy = groupBy != '' && groupBy != null;
+  const isCountFn = aggFn === AggFn.Count;
+  const groupByField =
+    hasGroupBy &&
+    buildSearchColumnName(propertyTypeMappingsModel.get(groupBy), groupBy);
+
+  const serializer = new SQLSerializer(propertyTypeMappingsModel);
+
+  const label = SqlString.escape(`${aggFn}(${field})`);
+
+  const selectClause = [
+    isCountFn
+      ? 'toFloat64(count()) as data'
```

Comment:
do we want to do this for other counts ?

Reply:
Aghhh good catch